s3pypi/core.py

```diff
@@ -100,3 +108,15 @@ def extract_wheel_metadata(path: Path) -> PackageMetadata:
             raise S3PyPiError(f"No wheel metadata found in {path}") from None
 
     return email.message_from_string(text)
+
+
+def calculate_hash(local_path: Path, hash_name: str):
```

Comment:
What do you think of putting an [`@lru_cache(maxsize=None)`](https://docs.python.org/3/library/functools.html#functools.lru_cache) on this function? That would prevent the hash from being recalculated on consecutive calls to the `filename` property above.

Reply:
It makes sense. Done